Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -121,9 +123,23 @@ public virtual void Initialize(IAlgorithm algorithm, IBrokerage brokerage, IResu
         /// </summary>
         public bool Ready
         {
-            get { return _orderQueue.Count == 0 && !_algorithm.ProcessingOrder; }
+            get
+            {
+                return HasPendingItems == false
+                    && !_algorithm.ProcessingEvents;
+            }
         }
 
+        public bool HasPendingItems
+        {
+            get
+            {
+                return _orderRequestQueue.Count > 0
+                    || _orderEventQueue.Count > 0
+                    || _securityEventQueue.Count > 0
+                    || _accountEventQueue.Count > 0;
```

Comment:
As per the [MSDN documentation](https://msdn.microsoft.com/en-us/library/dd267339(v=vs.110).aspx) the preferred way to check if a `ConcurrentQueue` has any items is to use the `IsEmpty` property.